server/src/test/scala/services/crunch/VoyageManifestsSpec.scala

```diff
@@ -115,155 +111,155 @@ class VoyageManifestsSpec extends CrunchTestLike {
-          queues = Map("T1" -> Seq(EeaDesk, EGate, NonEeaDesk))
-        ),
-        initialPortState = Option(PortState(Map(flight.uniqueId -> ApiFlightWithSplits(flight, Set())), Map(), Map()))
-      )
-
-      offerAndWait(crunch.manifestsInput, inputManifests)
-
-      val expected = Map(Queues.EeaDesk -> 1.0, Queues.EGate -> 1.0, Queues.NonEeaDesk -> 2.0)
-
-      crunch.liveTestProbe.fishForMessage(10 seconds) {
-        case ps: PortState =>
-          val queuePax = ps.crunchMinutes
-            .values
-            .filter(cm => cm.minute == SDate(scheduled).millisSinceEpoch)
-            .map(cm => (cm.queueName, cm.paxLoad))
-            .toMap
-
-          println(s"queuePax: $queuePax")
-
-          queuePax == expected
-      }
-
-      true
+    true
+  }
+
+  "Given a VoyageManifest with 2 transfers, 1 Eea Passport, 1 Eea Id card, and 2 visa nationals " +
+    "When I crunch the flight with 4 non-transit pax (10 pax minus 6 transit) " +
+    "Then I should see the 4 non-transit pax go to egates (1), eea desk (1), and non-eea (2)" >> {
+
+    val scheduled = "2017-01-01T00:00Z"
+    val portCode = "LHR"
+
+    val flight = ArrivalGenerator.apiFlight(flightId = Option(1), origin = "JFK", schDt = scheduled, iata = "BA0001", terminal = "T1", actPax = Option(10), tranPax = Option(6))
+    val inputManifests = ManifestsFeedSuccess(DqManifests("", Set(
+      VoyageManifest(DqEventCodes.CheckIn, portCode, "JFK", "0001", "BA", "2017-01-01", "00:00", List(
+        inTransitFlag,
+        inTransitCountry,
+        euPassport,
+        euIdCard,
+        visa,
+        visa
+      ))
+    )))
+    val crunch = runCrunchGraph(
+      now = () => SDate(scheduled),
+      airportConfig = airportConfig.copy(
+        portCode = portCode,
+        defaultProcessingTimes = Map("T1" -> Map(
+          eeaMachineReadableToDesk -> 25d / 60,
+          eeaNonMachineReadableToDesk -> 25d / 60,
+          eeaMachineReadableToEGate -> 25d / 60,
+          visaNationalToDesk -> 25d / 60
+        )),
+        terminalNames = Seq("T1"),
+        queues = Map("T1" -> Seq(EeaDesk, EGate, NonEeaDesk))
+      ),
+      initialPortState = Option(PortState(Map(flight.uniqueId -> ApiFlightWithSplits(flight, Set())), Map(), Map()))
+    )
+
+    offerAndWait(crunch.manifestsInput, inputManifests)
+
+    val expected = Map(Queues.EeaDesk -> 1.0, Queues.EGate -> 1.0, Queues.NonEeaDesk -> 2.0)
+
+    crunch.liveTestProbe.fishForMessage(10 seconds) {
+      case ps: PortState =>
+        val queuePax = ps.crunchMinutes
+          .values
+          .filter(cm => cm.minute == SDate(scheduled).millisSinceEpoch)
+          .map(cm => (cm.queueName, cm.paxLoad))
+          .toMap
+
+        println(s"queuePax: $queuePax")
+
+        queuePax == expected
     }
 
-    "Given initial VoyageManifests and no updates from the feed " +
-      "When I send in a live arrival " +
-      "Then I should see the flight with its API split in the port state" >> {
+    true
```

Comment:
replace `true` with `success` to avoid compiler warnings

Reply:
sick